### ISSUE TITLE
Build errors on KVM ARM 

### ISSUE THREAD

**Issue by razvand** [unikraft/unikraft#335]
### Describe the bug

Recent updates to the Unikraft core are causing build errors on KVM ARM.

This can be viewed on any recent PR, such as [this one](https://github.com/unikraft/unikraft/pull/333), shows the issue in the [CI/CD build output](https://builds.unikraft.io/builds/7222747).

### Steps to reproduce

Use a recent PR and build for KVM ARM.

### Expected behavior

The build should work without errors.

### Which architectures were you using or does this bug affect?

arm64

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

```shell
/tmp/build/80754af9/built/unikraft/plat/kvm/arm/lcpu.c: In function 'ukplat_lcpu_halt_irq':

/tmp/build/80754af9/built/unikraft/plat/kvm/arm/lcpu.c:49:2: warning: implicit declaration of function 'halt' [-Wimplicit-function-declaration]

   49 |  halt();

      |  ^~~~

/tmp/build/80754af9/built/unikraft/plat/common/arm/pci_bus_arm64.c: In function 'arch_pci_probe':

/tmp/build/80754af9/built/unikraft/plat/common/arm/pci_bus_arm64.c:99:5: warning: 'ret' may be used uninitialized in this function [-Wmaybe-uninitialized]

   99 |  if (ret < 0) {

      |     ^

/tmp/build/80754af9/built/unikraft/plat/common/arm/pci_bus_arm64.c:72:6: note: 'ret' was declared here

   72 |  int ret;

      |      ^~~

/tmp/build/80754af9/built/unikraft/lib/nolibc/stdio.c: In function 'fputc':

/tmp/build/80754af9/built/unikraft/lib/nolibc/stdio.c:497:5: warning: 'ret' may be used uninitialized in this function [-Wmaybe-uninitialized]

  497 |  if (ret == 1)

      |     ^

In file included from /tmp/build/80754af9/built/unikraft/include/uk/arch/tls.h:36,

                 from /tmp/build/80754af9/built/unikraft/lib/uksched/sched.c:39:

/tmp/build/80754af9/built/unikraft/arch/arm/arm64/include/uk/asm/tls.h:37:2: warning: #warning Thread-local storage has not been tested on aarch64! [-Wcpp]

   37 | #warning Thread-local storage has not been tested on aarch64!

      |  ^~~~~~~

In file included from /tmp/build/80754af9/built/unikraft/include/uk/arch/tls.h:36,

                 from /tmp/build/80754af9/built/unikraft/lib/uksched/thread.c:42:

/tmp/build/80754af9/built/unikraft/arch/arm/arm64/include/uk/asm/tls.h:37:2: warning: #warning Thread-local storage has not been tested on aarch64! [-Wcpp]

   37 | #warning Thread-local storage has not been tested on aarch64!

      |  ^~~~~~~

/tmp/build/80754af9/built/apps/helloworld/build/libsyscall_shim/uk_syscall.c: In function 'uk_vsyscall':

/tmp/build/80754af9/built/apps/helloworld/build/libsyscall_shim/uk_syscall.c:135:7: error: 'SYS_readlink' undeclared (first use in this function); did you mean 'SYS_readlinkat'?

  135 |  case SYS_readlink:

      |       ^~~~~~~~~~~~

      |       SYS_readlinkat

/tmp/build/80754af9/built/apps/helloworld/build/libsyscall_shim/uk_syscall.c:135:7: note: each undeclared identifier is reported only once for each function it appears in

/tmp/build/80754af9/built/apps/helloworld/build/libsyscall_shim/uk_syscall.c:148:7: error: 'SYS_link' undeclared (first use in this function); did you mean 'SYS_linkat'?

  148 |  case SYS_link:

      |       ^~~~~~~~

      |       SYS_linkat
```

**Comment by nderjung** [comment]
Duplicate of #330.
